File: CLASE I - Respaso/clase_1_ayed/carton.c


#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <time.h>

#include "carton.h"





CartonPtr crearCartonTeclado(); //Comprador seleccione los numeros

CartonPtr crearCarton2(int v[8], char d[15]);

CartonPtr crearCarton1(char d[15]){

    CartonPtr c = (CartonPtr) malloc(sizeof(struct CartonE));

    strcpy(c->direccion, d);

    //cargar vector cargarVector(c->vec);

    for ( int i = 0 ; i<8; i++){

        c->vec[i] = rand()%100; //0 a 99
        //Verificar que no se repitan!!!!

    }

    return c;

};

void destruirCarton(CartonPtr c);

void mostrarCarton(CartonPtr c){

    printf("\n\t------ CARTON ----\n");
    printf("\tDIRECCION: %s\n\t---VECTOR---\n\t", c->direccion);
    //mostrarVector
    for (int i =0 ; i<8; i++){

        printf(" %d ", c->vec[i]);

    }



};

File: CLASE I - Respaso/clase_1_ayed/golang/carton.go
package main

import (
	"fmt"
	"math/rand"
)

type Carton struct {
	vec       [8]int
	direccion string
}

func crearCarton1(direccion string) *Carton {
	c := &Carton{direccion: direccion}
	for i := 0; i < 8; i++ {
		c.vec[i] = rand.Intn(100) // 0..99
		// En el C original: falta verificar repetidos; se replica el comportamiento.
	}
	return c
}

func mostrarCarton(c *Carton) {
	fmt.Printf("\n\t------ CARTON ----\n")
	fmt.Printf("\tDIRECCION: %s\n\t---VECTOR---\n\t", c.direccion)
	for i := 0; i < 8; i++ {
		fmt.Printf(" %d ", c.vec[i])
	}
}

